src/libfsm/print/ir.c

```diff
@@ -81,7 +81,7 @@ static struct ir_group *
 make_groups(const struct fsm *fsm, const struct fsm_state *state, const struct fsm_state *mode,
 	size_t *u)
 {
-	struct range ranges[UCHAR_MAX]; /* worst case */
+	struct range ranges[UCHAR_MAX + 1]; /* worst case */
```

Comment:
Given that you've updated this in a few places, it seems like an enum constant might be safer here.  eg: 
```enum { MAX_RANGES = UCHAR_MAX + 1 };```

Reply:
I might do that sometime, but perhaps after I'm clearer about what the alphabet (Sigma) is for FSMs.